test/integration/sonobuoy_integration_test.go

```diff
@@ -0,0 +1,92 @@
+// +build integration
+
+package integration
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/heptio/sonobuoy/pkg/buildinfo"
+)
+
+var (
+	// Path to the Sonobuoy CLI
+	sonobuoy string
+
+	// Details from the git repository that should be available in the Sonobuoy version information
+	gitSHA     string
+	gitVersion string
+)
+
+func findSonobuoyCLI() (string, error) {
+	sonobuoyPath := os.Getenv("SONOBUOY_CLI")
+	if sonobuoyPath == "" {
+		sonobuoyPath = "../../sonobuoy"
+	}
+	if _, err := os.Stat(sonobuoyPath); os.IsNotExist(err) {
+		return "", err
+	}
+
+	return sonobuoyPath, nil
+}
+
+func TestSonobuoyVersion(t *testing.T) {
+	var stdout, stderr bytes.Buffer
+
+	command := exec.Command(sonobuoy, "version")
+	command.Stdout = &stdout
+	command.Stderr = &stderr
+
+	t.Logf("Running %q\n", command.String())
+	if err := command.Run(); err != nil {
+		t.Errorf("Sonobuoy exited with an error: %v", err)
+		t.Log(stderr.String())
+		t.FailNow()
+	}
+
+	expectedOutput := []string{
+		fmt.Sprintf("Sonobuoy Version: %v", gitVersion),
+		fmt.Sprintf("MinimumKubeVersion: %v", buildinfo.MinimumKubeVersion),
+		fmt.Sprintf("MaximumKubeVersion: %v", buildinfo.MaximumKubeVersion),
+		fmt.Sprintf("GitSHA: %v", gitSHA),
+	}
+
+	lines := strings.Split(stdout.String(), "\n")
+	for i, line := range lines {
+		if i < len(expectedOutput) {
+			if expectedOutput[i] != line {
+				t.Errorf("unexpected output in version information, expected %v, got %v", expectedOutput[i], line)
+			}
+		}
+	}
+}
+
+func TestMain(m *testing.M) {
+	var err error
+	sonobuoy, err = findSonobuoyCLI()
+	if err != nil {
+		fmt.Printf("Skipping integration tests: failed to find sonobuoy CLI: %v\n", err)
+		os.Exit(1)
+	}
+
+	flag.StringVar(&gitSHA, "git-sha", os.Getenv("GIT_SHA"), "SHA of HEAD Git commit (git rev-parse --verify HEAD)")
```

Comment:
nit: Can we define all these env vars at the top in a const section? That way we can know what env vars may impact the test run?